src/helpers/accessiblity.ts

```diff
@@ -233,3 +233,34 @@ export function isElementSelected(
   const { accessibilityState, 'aria-selected': ariaSelected } = element.props;
   return ariaSelected ?? accessibilityState?.selected ?? false;
 }
+
+export function getAccessibleName(
+  element: ReactTestInstance
+): string | undefined {
+  const labelTextFromLabel = getAccessibilityLabel(element);
+  const labelTextFromLabelledBy = getAccessibilityLabelledBy(element);
+
+  if (labelTextFromLabelledBy) {
+    const parentElement = getHostParent(element);
+
+    if (parentElement) {
+      const labelElement = parentElement.findByProps({
+        nativeID: labelTextFromLabelledBy,
+      });
+
+      if (labelElement) {
+        // Extract and return the text content of the label element
+        const label = labelElement.props.children;
+        return label;
+      }
+    }
+  }
+
+  if (labelTextFromLabel) {
+    return labelTextFromLabel;
+  }
+
+  const ownText = element?.children?.join(' ');
```

Comment:
There is a `getTextContent` fn already avialable in RNTL that would do that for you.